marshal/consumer.go

```diff
@@ -297,6 +305,32 @@ func (c *Consumer) claimPartitions() {
 	}
 }
 
+func (c *Consumer) topicsClaimed() map[string]bool {
+	claimed := make(map[string]bool)
+	for topic, topicClaims := range c.claims {
+		if topicClaims[0].Claimed() {
+			claimed[topic] = true
+		}
+	}
+
+	return claimed
+}
+
+func (c *Consumer) isTopicClaimReached(topic string) bool {
```

Comment:
The main reason is because you should allow the current topic partitions to be claimed if this topic has already been claimed by claiming partition 0...the check is also used inside tryClaimPartition, which claims any partition potentially..